tsl/src/remote/dist_ddl.c

```diff
@@ -978,6 +1000,10 @@ dist_ddl_start(ProcessUtilityArgs *args)
 	if (dist_ddl_scheduled_for_execution())
 		return;
 
+	/* Process remote DDL only on the distributed database */
+	if (!IsTransactionState() || dist_util_membership() == DIST_MEMBER_NONE)
```

Comment:
Two questions:
- why is this called for a non-distributed database at all?
- what does the `!IsTransactionState()` check mean?

Reply:
1) We have checks which block DDL operations executed directly on data nodes from a client. Basically we have a guc which allows or blocks those operations if they are not executed from the access node connection. This cases are handled there. So currently globally we prevent only non-distributed databases.

2) This is something I stood today. Since the dist ddl code executed after being already executed from main process utility hook in `src/process_utility.c` we could end in state, when transaction is already completed by `ROLLBACK` statement. Since `dist_util_membership()` function reads meta data catalog, it is required to have an active transaction there.

Reply:
Thanks. Maybe it makes sense to put this entire comment into code, I'll forget it again :)